sdk/communication/Azure.Communication.Common/src/PhoneNumberIdentifier.cs

```diff
@@ -27,17 +35,17 @@ public PhoneNumberIdentifier(string phoneNumber, string rawId = null)
         {
             Argument.AssertNotNullOrEmpty(phoneNumber, nameof(phoneNumber));
             PhoneNumber = phoneNumber;
-            RawId = rawId;
+            _rawId = rawId;
```

Comment:
I prefer to have as little logic in the constructor for several reasons. E.g., not to waste processor time on things that might not be needed at the time of construction. In this case, it can be negligible but for the `MicrosoftTeamsUserIdentifier` it's already four if-elses. I changed the implementation though to support lazy initialization of the properties so that they don't need to be recalculated every time (https://github.com/Azure/azure-sdk-for-net/pull/28574/commits/9bd080b83af940120e1989eb872ec8f7c4f9afe7).

Please let me know, if it makes sense.

If this is not idiomatic for Java, feel free to do the stuff in the constructor.